src/appleseed/renderer/kernel/lighting/pathtracer.h

```diff
@@ -289,6 +292,102 @@ size_t PathTracer<PathVisitor, Adjoint>::trace(
             vertex.m_bsdf_data = bsdf_input_evaluator.data();
         }
 
+        // Evaluate the input values of the BSSRDF and sample it.
+        InputEvaluator bssrdf_input_evaluator(shading_context.get_texture_cache());
+        if (vertex.m_bssrdf)
+        {
+            vertex.m_bssrdf->evaluate_inputs(
+                shading_context,
+                bssrdf_input_evaluator,
+                *vertex.m_shading_point);
+            vertex.m_bssrdf_data = bssrdf_input_evaluator.data();
+
+            BSSRDFSample sample(*vertex.m_shading_point, vertex.m_sampling_context);
+            if (vertex.m_bssrdf->sample(vertex.m_bssrdf_data, sample))
+            {
+                vertex.m_bssrdf_eta = sample.get_eta();
+                vertex.m_bssrdf_outgoing_fresnel =
+                    foundation::fresnel_transmission(
+                        foundation::dot(
+                            vertex.m_outgoing.get_value(),
+                            vertex.get_shading_normal()),
+                        vertex.m_bssrdf_eta);
+
+                if (vertex.m_bssrdf_outgoing_fresnel == 0.0)
+                    vertex.m_bssrdf = 0;
+            }
+            else
+                vertex.m_bssrdf = 0;
+
+            if (vertex.m_bssrdf)
```

Comment:
You could save one `if` in both the SSS and non-SSS cases by moving the content of this `if` under the `if (vertex.m_bssrdf_outgoing_fresnel == 0.0)` above.